File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/integration/keyos/KeyOsWidevineContentProtectionIntegrationFactory.java
package com.theoplayer.contentprotectionintegration.integration.keyos;

import com.theoplayer.android.api.contentprotection.ContentProtectionIntegration;
import com.theoplayer.android.api.contentprotection.ContentProtectionIntegrationFactory;
import com.theoplayer.android.api.source.drm.DRMConfiguration;
import com.theoplayer.contentprotectionintegration.integration.keyos.KeyOsWidevineContentProtectionIntegration;

public class KeyOsWidevineContentProtectionIntegrationFactory implements ContentProtectionIntegrationFactory {
    @Override
    public ContentProtectionIntegration build(DRMConfiguration configuration) {
        return new KeyOsWidevineContentProtectionIntegration(configuration);
    }
}

File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/integration/keyos/KeyOsWidevineContentProtectionIntegrationFactory.kt
package com.theoplayer.contentprotectionintegration.integration.keyos

import com.theoplayer.android.api.contentprotection.ContentProtectionIntegration
import com.theoplayer.android.api.contentprotection.ContentProtectionIntegrationFactory
import com.theoplayer.android.api.source.drm.DRMConfiguration

class KeyOsWidevineContentProtectionIntegrationFactory : ContentProtectionIntegrationFactory {
    override fun build(configuration: DRMConfiguration): ContentProtectionIntegration {
        return KeyOsWidevineContentProtectionIntegration(configuration)
    }
}